### PR TITLE
[SPARK-48924][PS] Add a pandas-like `make_interval` helper function

### PR DESCRIPTION
**Pull request by zhengruifeng**
### What changes were proposed in this pull request?
Add a pandas-like `make_interval` helper function

### Why are the changes needed?
factor it out as a helper function to be reusable

### Does this PR introduce _any_ user-facing change?
No, internal change only

### How was this patch tested?
CI

### Was this patch authored or co-authored using generative AI tooling?
No

### CODE REVIEW

File: python/pyspark/pandas/resample.py [hunk 1]
```diff
@@ -130,19 +130,6 @@ def _resamplekey_type(self) -> DataType:
     def _agg_columns_scols(self) -> List[Column]:
         return [s.spark.column for s in self._agg_columns]
 
-    def get_make_interval(  # type: ignore[return]
```
Comment (xinrong-meng) [inline python/pyspark/pandas/resample.py:133]: Just wondering if we skip some units intentionally here

**Review by xinrong-meng — APPROVED** [review]

**Comment by xinrong-meng** [comment]
LGTM thank you!

File: python/pyspark/pandas/resample.py (hunk 1 above)
Comment (zhengruifeng) [reply python/pyspark/pandas/resample.py:133]: some units was ignored due to not invoked at that time, I am adding them back to make the helper function cover all pandas units :)

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

File: python/pyspark/pandas/resample.py (hunk 1 above)
Comment (xinrong-meng) [reply python/pyspark/pandas/resample.py:133]: Great thanks!

**Pull request closed without merging**